distributed/scheduler.py

```diff
@@ -367,11 +348,42 @@ def identity(self, stream):
              'workers': dict(self.worker_info)}
         return d
 
-    def get_versions(self, stream):
+    def get_worker_service_addr(self, worker, service_name):
+        """
+        Get the (host, port) address of the named service on the *worker*.
+        Returns None if the service doesn't exist.
+        """
+        info = self.worker_info[worker]
+        port = info['services'].get(service_name)
+        if port is None:
+            return None
+        else:
+            return info['host'], port
+
+    def get_versions(self, comm):
         """ Basic information about ourselves and our cluster """
         return get_versions()
 
-    def start(self, port=8786, start_queues=True):
+    def start_services(self, listen_ip=''):
+        for k, v in self.service_specs.items():
+            if isinstance(k, tuple):
+                k, port = k
+            else:
+                port = 0
+
+            try:
+                service = v(self, io_loop=self.loop)
+                service.listen((listen_ip, port))
+                self.services[k] = service
+            except Exception as e:
+                logger.info("Could not launch service: %r", (k, port),
+                            exc_info=True)
+
+    def stop_services(self):
+        for service in self.services.values():
+            service.stop()
```

Comment:
We might eventually be able to unify this on `Server`.  Both Scheduler and Worker should have similar needs here.